pkg/core/xds/rules.go

```diff
@@ -39,7 +39,7 @@ func (ss Subset) IsSubset(other Subset) bool {
 			if otherTag.Value == tag.Value && otherTag.Not != tag.Not {
 				return false
 			}
-			if otherTag.Value != tag.Value && !otherTag.Not {
+			if otherTag.Value != tag.Value && otherTag.Not == tag.Not {
```

Comment:
It works for us because `otherTag.Not` is always `false` (we don't allow "not" in policies, only in rules). But in general, I think it's still an incorrect condition:
```
other:
  key1: !value1
---
tag:
  key1: !value2
```
by looking at `key1` we can't say for sure if `other` is (or isn't) a subset of `tag`. 

So probably the condition should be:
```go
if otherTag.Value == tag.Value && otherTag.Not != tag.Not {
	return false
}
if otherTag.Value != tag.Value && !otherTag.Not && !tag.Not {
	return false
}
```

Reply:
Ok, make sense. Changed.